File: set_1/challenge_1.go
package main

import "fmt"
import "encoding/hex"
import "encoding/base64"

func main() {
  hexString :=
  "49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d"
  rawBytes, err := hex.DecodeString(hexString)
  if err != nil {
    fmt.Println("Unable to convert hex to byte. ", err)
  }
  base64String := base64.StdEncoding.EncodeToString(rawBytes)
  fmt.Println(base64String)
}

File: set_1/challenge_1.rb
def hex_to_base64(hex)
  base64_table = [*'A'..'Z', *'a'..'z', *'0'..'9', '+', '/']
  bytes = hex.hex.digits(64).reverse
  base64 = bytes.map { |byte| base64_table[byte] }.join
end

hex = "49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d"
base64 = hex_to_base64(hex)
puts base64
